tiatoolbox/tools/registration/wsi_registration.py

```diff
@@ -0,0 +1,32 @@
+import numpy as np
+from skimage import exposure, filters, morphology
+
+
+def preprocess(fixed_img, moving_img):
+    """Performs histogram normalization between two images.
+
+    Args:
+        fixed_img (:class:`numpy.ndarray`):
+            A grayscale fixed image
+        moving_img (:class:`numpy.ndarray`):
+            A grayscale moving image
+
+    Returns:
+        :class:`numpy.ndarray`:
+            A normalized grayscale fixed image
+        :class:`numpy.ndarray`:
+            A normalized grayscale moving image
+
+    """
+    if len(fixed_img.shape) != 2 or len(moving_img.shape) != 2:
+        raise ValueError("The input images should be grayscale images.")
+
+    moving_entropy, fixed_entropy = filters.rank.entropy(
+        moving_img, morphology.disk(3)
+    ), filters.rank.entropy(fixed_img, morphology.disk(3))
+    if np.mean(fixed_entropy) > np.mean(moving_entropy):
+        moving_img = exposure.match_histograms(moving_img, fixed_img)
+    else:
+        fixed_img = exposure.match_histograms(fixed_img, moving_img)
+
+    return fixed_img, moving_img
```

Comment:
I would suggest changing the names from fixed and moving to image a and image a because there is no code specific here to the images being fixed or moving. It is simply matching the histogram of one image to another.